### PR TITLE
linting: fix issues in telemeter, add linting to CI

### PR DESCRIPTION
**Pull request by s-urbaniak**

### CODE REVIEW

**Comment by s-urbaniak** [comment]
@squat PTAL, this is ready for a pass :-)

File: Makefile [hunk 1]
```diff
@@ -17,7 +19,12 @@ build:
 image:
 	imagebuilder -t openshift/telemeter:latest .
 
-check:
+lint: $(GOLANGCI_LINT_BIN)
+	# megacheck fails to respect build flags, causing compilation failure during linting.
+	# instead, use the unused, gosimple, and staticcheck linters directly
+	$(BIN)/golangci-lint run -e '.*Write' -D megacheck -E unused,gosimple,staticcheck
```
Comment (s-urbaniak) [inline Makefile]: honestly I am not sure about ignoring `.*Write` method signatures not checking errors. We can enable this, the code just becomes a bit chatty in some places with `_ = Write...`.

File: pkg/authorize/tollbooth/tollbooth.go [hunk 2]
```diff
@@ -59,7 +59,7 @@ func (a *authorizer) AuthorizeCluster(token, cluster string) (string, error) {
 	defer func() {
 		// read the body to keep the upstream connection open
 		if resp.Body != nil {
-			io.Copy(ioutil.Discard, resp.Body)
+			_, _ = io.Copy(ioutil.Discard, resp.Body)
```
Comment (squat) [inline pkg/authorize/tollbooth/tollbooth.go]: We can ignore the byte count but we shouldn’t suppress the error. If there is one, let’s at the very least log it

File: pkg/metricsclient/metricsclient.go [hunk 3]
```diff
@@ -118,13 +118,13 @@ func (c *Client) Send(ctx context.Context, req *http.Request, families []*client
...
 
-	ctx, cancel := context.WithTimeout(context.Background(), c.timeout)
+	ctx, cancel := context.WithTimeout(ctx, c.timeout)
 	req = req.WithContext(ctx)
 	defer cancel()
 
 	return withCancel(ctx, c.client, req, func(resp *http.Response) error {
 		defer func() {
-			io.Copy(ioutil.Discard, resp.Body)
+			_, _ = io.Copy(ioutil.Discard, resp.Body)
```
Comment (squat) [inline pkg/metricsclient/metricsclient.go]: Same here

File: Makefile (hunk 1 above)
Comment (squat) [reply Makefile]: I don’t much like the idea of ignoring the errors in `.Write...`. We can ignore the byte counts but even if we don’t return the error, at least log it. Go2’s error handling will make this feel less “chatty” 😉 the insight into write failures can be very informative about network, memory, or disk issues.

**Comment by s-urbaniak** [comment]
@squat addressed comments, PTAL

File: pkg/fnv/hash.go [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+import (
+	"fmt"
+	"hash"
+	"hash/fnv"
+	"strconv"
+)
+
+// Hash hashes the given text using a 64-bit FNV-1a hash.Hash.
+func Hash(text string) (string, error) {
```
Comment (squat) [inline pkg/fnv/hash.go]: These hash changes are related to the comment [0] in the refactoring PR but I think they are out of scope for this linting PR. Let's move this into it's separate pull request so we can keep the PR as one cohesive unit of work centered around linting.

[0] https://github.com/openshift/telemeter/pull/47#discussion_r225924456

Comment (s-urbaniak) [reply pkg/fnv/hash.go]: oki, agreed

**Comment by s-urbaniak** [comment]
@squat removed the commit, PTAL
